flows/engine/assets.go

```diff
@@ -3,90 +3,147 @@ package engine
 import (
 	"encoding/json"
 	"fmt"
+	"sync"
+	"time"
 
 	"github.com/nyaruka/goflow/flows"
 	"github.com/nyaruka/goflow/flows/definition"
 	"github.com/nyaruka/goflow/utils"
 )
 
-type assets struct {
-	flows    map[flows.FlowUUID]flows.Flow
-	channels map[flows.ChannelUUID]flows.Channel
+type assetContainer struct {
+	assetType  flows.AssetType
+	asset      flows.Asset
+	accessedOn *time.Time
+	expiresOn  time.Time
+	fetchURL   string
 }
 
-func NewAssets(flowList []flows.Flow, channelList []flows.Channel) flows.Assets {
-	flowMap := make(map[flows.FlowUUID]flows.Flow, len(channelList))
-	for _, flow := range flowList {
-		flowMap[flow.UUID()] = flow
+type assetManager struct {
+	cache      map[flows.AssetUUID]assetContainer
+	cacheMutex sync.Mutex
+}
+
+func NewAssetManager() flows.AssetManager {
```

Comment:
Not passionate, but I think I prefer just `Assets` over `AssetManager`.. `Manager` always smells a bit as to me it doesn't carry a lot of meaning with it.